app/models/bulk_download.rb

```diff
@@ -23,6 +23,47 @@ class BulkDownload < ApplicationRecord
   attr_accessor :params
 
   validates :status, presence: true, inclusion: { in: [STATUS_WAITING, STATUS_RUNNING, STATUS_ERROR, STATUS_SUCCESS] }
+  validate :params_checks
+
+  def params_checks
+    set_params
+    params = self.params
+
+    params_value = lambda do |key|
+      return nil if params.nil?
+      # Each field param should be structured as
+      # { displayName: "FOO", value: "BAR" }
+      # field_params[key] should be ActionController::Parameters object that responds to :keys
+      params[key].respond_to?(:keys) ? params[key]["value"] : nil
```

Comment:
Generally: Isn't in better to just check if it is an Hash or subclass: `obj.is_a?(Hash)`? 
In theory, this accepts any object that has a keys method even if it does not do what you expect...

The params inside params are also ActionController::Parameters?

Reply:
Deleted this method.